File: software/examples/rust/example_threshold.rs
use std::{error::Error, io, thread};
use tinkerforge::{ip_connection::IpConnection, uv_light_v2_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your UV Light Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let uvl = UvLightV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    let uvi_receiver = uvl.get_uvi_callback_receiver();

    // Spawn thread to handle received callback messages.
    // This thread ends when the `uvl` object
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for uvi in uvi_receiver {
            println!("UV Index: {}", uvi as f32 / 10.0);
            println!("UV index > 3. Use sunscreen!");
        }
    });

    // Configure threshold for UV index "greater than 3"
    // with a debounce period of 1s (1000ms).
    uvl.set_uvi_callback_configuration(1000, false, '>', 3 * 10, 0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/uv_light_v2_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your UV Light Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	uvl, _ := uv_light_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	uvl.RegisterUVICallback(func(uvi int32) {
		fmt.Printf("UV Index: %f\n", float64(uvi)/10.0)
		fmt.Println("UV index > 3. Use sunscreen!")
	})

	// Configure threshold for UV index "greater than 3"
	// with a debounce period of 1s (1000ms).
	uvl.SetUVICallbackConfiguration(1000, false, '>', 3*10, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
